frontend/src/Indexer/Edit/EditIndexerModalContent.js

```diff
@@ -156,6 +158,23 @@ function EditIndexerModalContent(props) {
                 />
               </FormGroup>
 
+              <FormGroup
```

Comment:
I thought about two different ways of achieving this.

1. Omit `downloadClientId`. for example:

```
// EditIndexerModalContentConnector.js
function createMapStateToProps() {
  return createSelector(
    (state) => state.settings.advancedSettings,
    (state) => state.settings.downloadClients,
    createIndexerSchemaSelector(),
    (advancedSettings, downloadClients, indexer) => {
      if (downloadClients.items.length) {
        return {
          advancedSettings,
          ...indexer
        };
      }

      const { downloadClientId, ...indexerWithoutDownloadClientId } = indexer;
      return {
        advancedSettings,
        ...indexerWithoutDownloadClientId
      };
    });
}

// EditIndexerModalContent.js
              {downloadClientId && (
                <FormGroup
                  advancedSettings={advancedSettings}
                  isAdvanced={true}
                >
                  <FormLabel>{translate('DownloadClient')}</FormLabel>

                  <FormInputGroup
                    type={inputTypes.DOWNLOAD_CLIENT_SELECT}
                    name="downloadClientId"
                    helpText={translate('IndexerDownloadClientHelpText')}
                    {...downloadClientId}
                    includeAny={true}
                    protocol={protocol.value}
                    onChange={onInputChange}
                  />
                </FormGroup>
              )}

```

2. Pass down a new `downloadClientsCount` prop or something similar to EditIndexerModalContent.

Reply:
It's worth mentioning that for this to work properly we would need to call `fetchDownloadClients` in PageConnector, otherwise `downloadClients.items` is always empty unless it has been fetched somewhere else, like for example in the download client settings.